tests/lax_numpy_indexing_test.py

```diff
@@ -849,6 +849,11 @@ def testTrivialGatherIsntGenerated(self):
     self.assertEqual(len(jaxpr.jaxpr.eqns), 1)
     self.assertNotIn('gather', str(jaxpr))
 
+    jaxpr = jax.make_jaxpr(lambda x: x[0:6:1])(np.arange(4))
+    self.assertEqual(len(jaxpr.jaxpr.eqns), 0)
+    jaxpr = jax.make_jaxpr(lambda x: x[:4])(np.arange(4))
+    self.assertEqual(len(jaxpr.jaxpr.eqns), 0)
```

Comment:
I thought about parametrizing the test but it turned out to be much more readable this way with some slight duplication.